scripts/update_licenses.py

```diff
@@ -0,0 +1,59 @@
+import json
+import os
+import re
+from pathlib import Path
+from github import Github
+
+CURRENT_FILE = Path(__file__)
+ROOT = CURRENT_FILE.parents[1]
+GITHUB_TOKEN = os.getenv("GITHUB_TOKEN", None)
+
+
+def main() -> None:
+    """
+    Script entry point.
+    """
+    repo = Github(login_or_token=GITHUB_TOKEN).get_repo("github/choosealicense.com")
+    license_dir = ROOT / "{{cookiecutter.project_slug}}" / "licenses"
+    titles = []
+    for file in repo.get_contents("_licenses", "gh-pages"):
+        content = file.decoded_content.decode(file.encoding)
+        titles.append(content.split("\n", maxsplit=2)[1].replace("title: ", ""))
+        path = license_dir / file.name
+        if not path.is_file():
+            (license_dir / file.name).write_text(replace_content_options(content))
+    # Put "Not open source" at front so people know it's an option
+    front_options = [
+        "Not open source",
+        "MIT License",
+        'BSD 3-Clause "New" or "Revised" License',
+        "GNU General Public License v3.0",
+        "Apache License 2.0",
+    ]
+    titles = [x for x in sorted(titles) if x not in front_options]
+    update_cookiecutter(front_options + titles)
+
+
+year = (re.compile(r"\[year]"), "{% now 'utc', '%Y' %}")
+email = (re.compile(r"\[email]"), "{{ cookiecutter.email }}")
+fullname = (re.compile(r"\[fullname]"), "{{ cookiecutter.author_name }}")
+project = (re.compile(r"\[project]"), "{{ cookiecutter.project_name }}")
+projecturl = (re.compile(r"\[projecturl]"), "{{ cookiecutter.domain_name }}")
```

Comment:
Do you think we could add a test to make sure we don't miss any new placeholder? Seems a bit tricky to make the difference between a placeholder and a genuine `[...]` though. 🤔

Reply:
That's why I wanted the pull request initially... In the Jekyll headers for all these files, there are directions that specify what to replace though! The directions say something like "replace [year] with etc." if I'm not mistaken. So after replacing everything, if we still match a regex of `\[.*?\]` (which matches to anything with square brackets including if there are multiple square brackets like `[asdf] [asf]` will be two matches).

Side note: I think we don't really need to care about Appendix directions (like that found in Apache 2.0 where they say to add the boilerplate code with year and name. With us putting it in already, I think the developer will understand. Additionally, license years don't have to be updated so there's that...)